src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -146,8 +146,14 @@ export class PageControllerBase {
         if (isPageHeading) {
           labelOrLegend.isPageHeading = isPageHeading
 
+          // Check for optional in label
+          const isOptional =
+            this.components.formItems.at(0)?.options.required === false
```

Comment:
Is this the same as `formComponents[0]` above?

Sorry not checked properly! Thanks @KiranVarma02

```suggestion
            formComponents[0].options.required === false
```

Reply:
No, its not same. formComponents doesn't have options data.